### PR TITLE
polyfactory 2.17.0 :snowflake:

### PR DESCRIPTION
**Pull request by psteyer**
polyfactory 2.17.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5396]
- dev_url:        https://github.com/litestar-org/polyfactory/tree/v2.17.0
- conda_forge:    https://github.com/conda-forge/polyfactory-feedstock/blob/main/recipe/meta.yaml
- pypi:           https://pypi.org/project/polyfactory/2.17.0
- pypi inspector: https://inspector.pypi.io/project/polyfactory/2.17.0

### Explanation of changes:

- new version number

[PKG-5396]: https://anaconda.atlassian.net/browse/PKG-5396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,34 +10,67 @@ source:
...
     - pip
   run:
-    - python >=3.8,<4.0
+    - python
     - faker
     - typing-extensions
 
+{% set tests_to_skip = "" %} 
+{% set tests_to_ignore = "" %}
+# requires msgspec
```
Comment (boldorider4) [inline recipe/meta.yaml]: soon we may have `msgspec` by the way.
https://github.com/AnacondaRecipes/msgspec-feedstock/pull/1

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,34 +10,66 @@ source:
...
   host:
-    - python >=3.8,<4.0
+    - python 
     - hatchling
     - pip
   run:
-    - python >=3.8,<4.0
+    - python
     - faker
     - typing-extensions
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - typing-extensions >=4.6.0
```
see https://github.com/litestar-org/polyfactory/blob/v2.17.0/pyproject.toml#L43C22-L43C29

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,34 +10,64 @@ source:
...
-    - python >=3.8,<4.0
+    - python 
     - hatchling
     - pip
   run:
-    - python >=3.8,<4.0
+    - python
     - faker
-    - typing-extensions
+    - typing-extensions >=4.6.0
```
Comment (msentissi) [inline recipe/meta.yaml:25]: Would it make sense to add the pinned optional dependencies to a `run_constrained` section here ?
- sqlalchemy>=1.4.29
- odmantic<1.0.0
- attrs>=22.2.0
- email-validator>=2.0.0 (from pydantic[email])

Comment (psteyer) [reply recipe/meta.yaml:25]: I can add them all except:
- `email-validator 2.0.0` our highest version is `1.2.1`
- `odmantic` which we don't have

Comment (msentissi) [reply recipe/meta.yaml:25]: I don't think we need to have them to add them to `run_constrained`.

**Review by msentissi — APPROVED** [review]
LGTM!

**Pull request merged**